src/main/java/org/mule/tools/maven/plugin/module/analyze/DefaultModuleApiAnalyzer.java

```diff
@@ -51,80 +56,168 @@ public ProjectDependencyAnalysis analyze(MavenProject project, AnalyzerLogger an
-              if (packagesToAdd != null && !packagesToAdd.isEmpty()) {
-                Set<String> currentNoExportedPackageDeps = missingExportedPackages.get(missingExportedPackageDep);
-                if (currentNoExportedPackageDeps == null) {
-                  currentNoExportedPackageDeps = new HashSet<>();
-                  missingExportedPackages.put(missingExportedPackageDep, currentNoExportedPackageDeps);
+      final Map<String, Set<String>> externalPackageDeps = calculateExternalDeps(project, analyzerLogger);
+
+      Set<String> exportedPackageClosure = new HashSet<>(projectExportedPackages);
+      Set<String> missingAnalyzedPackages = new HashSet<>();
+      final Set<String> jrePackages = find();
+
+      boolean dirty;
+      do {
+        Set<String> diff = new HashSet<>();
+        Set<String> exportedByOtherModule = new HashSet<>();
+        for (String exportedPackage : exportedPackageClosure) {
+          if (!ignorePackage(analyzerLogger, exportedPackage, jrePackages, externalExportedPackages)) {
+            Set<String> packageDeps = projectPackageDependencies.get(exportedPackage);
+            if (packageDeps != null) {
+              for (String packageDep : packageDeps) {
+                if (!exportedPackageClosure.contains(packageDep)
+                    && !ignorePackage(analyzerLogger, packageDep, jrePackages, externalExportedPackages)) {
+                  diff.add(packageDep);
                 }
-                for (String packageToAdd : packagesToAdd) {
-                  if (!externalExportedPackages.contains(packageToAdd)) {
-                    if (currentNoExportedPackageDeps.add(packageToAdd)) {
-                      dirty = true;
-                    }
+              }
+            } else {
+              packageDeps = externalPackageDeps.get(exportedPackage);
+              if (packageDeps != null) {
+                for (String packageDep : packageDeps) {
+                  if (!exportedPackageClosure.contains(packageDep)
+                      && !ignorePackage(analyzerLogger, packageDep, jrePackages, externalExportedPackages)) {
+                    diff.add(packageDep);
                   }
                 }
+              } else {
+                if (!externalExportedPackages.contains(exportedPackage) && !projectOptionalPackages.contains(exportedPackage)) {
+                  missingAnalyzedPackages.add(exportedPackage);
+                }
               }
             }
           }
-        } while (dirty);
+        }
+        exportedPackageClosure.addAll(diff);
+        exportedPackageClosure.removeAll(exportedByOtherModule);
+        diff.removeAll(exportedByOtherModule);
+        dirty = !diff.isEmpty();
+      } while (dirty);
+
+      StringBuilder builder = new StringBuilder("Exported package closure:");
+      for (String exportedPackage : exportedPackageClosure) {
+        builder.append("\n").append(exportedPackage);
```

Comment:
Current log is displayed as:
[INFO] Found module: foo with property=artifact.export.classPackages:
org.foo
org.bar

with your proposal is displayed as:
[INFO] Found module: foo with property=artifact.export.classPackages:
	org.foo
	org.bar

I prefer the current version